plugins/kernel_v2/config/disk.rb

```diff
@@ -135,7 +140,18 @@ def validate(machine)
                            types: DEFAULT_DISK_TYPES.join(', '))
         end
 
-        if @disk_ext
+        if @disk_ext == UNSET_VALUE
+          # Work around to finalize disk_ext with a valid default per-provider
```

Comment:
Instead of putting provider logic here, it would be better to just have a default defined (in the case here `"vdi"`) and have a provider capability for `default_disk_ext` be checked to determine an acceptable default.